.make/docker.mk

```diff
@@ -0,0 +1,132 @@
+
+.PHONY: docker-image-builder
+## Builds the docker image used to build the software.
+docker-image-builder:
+	@echo "Building docker image $(DOCKER_IMAGE_CORE)"
+	docker build -t $(DOCKER_IMAGE_CORE) -f $(CUR_DIR)/Dockerfile.builder $(CUR_DIR)
+
+.PHONY: docker-image-deploy
+## Creates a runnable image using the artifacts from the bin directory.
+docker-image-deploy:
+	docker build -t $(DOCKER_IMAGE_DEPLOY) -f $(CUR_DIR)/Dockerfile.deploy $(CUR_DIR)
+
+.PHONY: docker-publish-deploy
+## Tags the runnable image and pushes it to the docker hub.
+docker-publish-deploy:
+	docker tag $(DOCKER_IMAGE_DEPLOY) fabric8-services/${PROJECT_NAME}:latest
+	docker push fabric8-services/${PROJECT_NAME}:latest
+
+.PHONY: docker-build-dir
+## Creates the docker build directory.
+docker-build-dir:
+	@echo "Creating build directory $(BUILD_DIR)"
+	mkdir -p $(DOCKER_BUILD_DIR)
+
+CLEAN_TARGETS += clean-docker-build-container
+.PHONY: clean-docker-build-container
+## Removes any existing container used to build the software (if any).
+clean-docker-build-container:
+	@echo "Removing container named \"$(DOCKER_CONTAINER_NAME)\" (if any)"
+ifneq ($(strip $(shell docker ps -qa --filter "name=$(DOCKER_CONTAINER_NAME)" 2>/dev/null)),)
+	@docker rm -f $(DOCKER_CONTAINER_NAME)
+else
+	@echo "No container named \"$(DOCKER_CONTAINER_NAME)\" to remove"
+endif
+
+CLEAN_TARGETS += clean-docker-build-dir
+.PHONY: clean-docker-build-dir
+## Removes the docker build directory.
+clean-docker-build-dir:
+	@echo "Cleaning build directory $(BUILD_DIR)"
+	-rm -rf $(DOCKER_BUILD_DIR)
+
+.PHONY: docker-start
+## Starts the docker build container in the background (detached mode).
+## After calling this command you can invoke all the make targets from the
+## normal Makefile (e.g. deps, generate, build) inside the build container
+## by prefixing them with "docker-". For example to execute "make deps"
+## inside the build container, just run "make docker-deps".
+## To remove the container when no longer needed, call "make docker-rm".
+docker-start: docker-build-dir docker-image-builder
+ifneq ($(strip $(shell docker ps -qa --filter "name=$(DOCKER_CONTAINER_NAME)" 2>/dev/null)),)
+	@echo "Docker container \"$(DOCKER_CONTAINER_NAME)\" already exists. To recreate, run \"make docker-rm\"."
+else
+	docker run \
+		--detach=true \
+		-t \
+		$(DOCKER_RUN_INTERACTIVE_SWITCH) \
+		--name="$(DOCKER_CONTAINER_NAME)" \
+		-v $(CUR_DIR):$(PACKAGE_PATH):Z \
+		-u $(shell id -u $(USER)):$(shell id -g $(USER)) \
+		-e GOPATH=$(GOPATH_IN_CONTAINER) \
+		-w $(PACKAGE_PATH) \
+		$(DOCKER_IMAGE_CORE)
+		@echo "Docker container \"$(DOCKER_CONTAINER_NAME)\" created. Continue with \"make docker-deps\"."
+endif
+
+.PHONY: build-linux
+## Build for linux
+build-linux: $(SOURCES)
+	GO15VENDOREXPERIMENT=1 GOARCH=amd64 GOOS=linux go build -o bin/fabric8-toggles-service-linux
+
+.PHONY: docker-image-deploylinux
+## Creates a runnable image using the artifacts from the bin directory.
+docker-image-deploy-linux:
```

Comment:
you could add a prerequisite on this goal: `bin/fabric8-toggles-service-linux`, to make sure the binary is built